### PR TITLE
Fixes #3 Added typeahead capability to add meal screen.

### LINKED ISSUE #3: Searching
Replace `<select>`s with typeahead searching https://twitter.github.io/typeahead.js/

### PR DESCRIPTION
**Pull request by kylec32**
Added the Twitter typeahead library and implemented it in the add meal modal.

### CODE REVIEW

File: grails-app/views/layouts/bootstrap.gsp [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <asset:stylesheet src="application.css" />
     <asset:stylesheet src="fullcalendar.print.css" media="print" />
     <asset:javascript src="application.js" />
+    <asset:javascript src="typeahead.js" />
```
Comment (mftruso) [inline grails-app/views/layouts/bootstrap.gsp]: Let's require this in `application.js` like the other scripts

File: grails-app/views/meal/list.gsp [hunk 2]
```diff
@@ -76,6 +78,10 @@
                     alert('Oops! There was a problem making this meal.');
                 });
 
+                $.each($(".dish"),function(index,field){
+                                    console.log($(field).val(""))
```
Comment (mftruso) [inline grails-app/views/meal/list.gsp]: Please fix the spacing here

File: grails-app/views/meal/list.gsp [hunk 3]
```diff
@@ -114,8 +125,17 @@
                     <g:hiddenField type="hidden" name="mealDate" />
 
                     <g:each in="${DishType.listOrderBySortOrder()}" var="dishType" >
-                        <label>${dishType.name}</label>
-                        <g:select name="${dishType.name.toLowerCase()}Dish" class="form-control dish" from="${Dish.findAllByType(dishType, [sort: 'name', order: 'asc'])}" optionValue="name" optionKey="id" noSelection="['':'Select a Dish']"  />
+                        <div>
+                            <label>${dishType.name}</label>
+                            <br/>
+
+                            <script>
```
Comment (mftruso) [inline grails-app/views/meal/list.gsp]: please use the `<asset:script>` tag here

File: grails-app/assets/stylesheets/application.css [hunk 4]
```diff
@@ -12,3 +12,73 @@
 *= require fullcalendar
 *= require_self
 */
+
+
+.typeahead,
```
Comment (mftruso) [inline grails-app/assets/stylesheets/application.css]: let's move this to a new file `main.css` and require that file here

File: grails-app/assets/javascripts/application.js [hunk 5]
```diff
@@ -3,4 +3,28 @@
 //= require moment
 //= require fullcalendar
 //= require_tree .
-//= require_self
\ No newline at end of file
+//= require_self
+
+function intializeTypeahead(id, type) {
```
Comment (mftruso) [inline grails-app/assets/javascripts/application.js]: let's move this to a new file called `main.js` and require that file here

**Review by mftruso — COMMENTED** [review]
Looks great! Just a few adjustments.

**Comment by kylec32** [comment]
Some valid points! Let me know if there is any other changes I can make to make this better. (2949635)

File: grails-app/assets/javascripts/main.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+function intializeTypeahead(id, type) {
+
+    $(document).ready(function(){
```
Comment (mftruso) [inline grails-app/assets/javascripts/main.js:3]: One more small thing, I don't think this `$(document).ready` is necessary

**Comment by mftruso** [comment]
Thanks!

**Pull request merged**